ReplyAdaptabilityCodelab/app/src/main/java/com/example/reply/ui/finish/ReplyApp.kt

```diff
@@ -0,0 +1,284 @@
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalNavigationDrawer
+import androidx.compose.material3.NavigationBar
+import androidx.compose.material3.NavigationBarItem
+import androidx.compose.material3.NavigationDrawerItem
+import androidx.compose.material3.NavigationDrawerItemDefaults
+import androidx.compose.material3.NavigationRail
+import androidx.compose.material3.NavigationRailItem
+import androidx.compose.material3.PermanentNavigationDrawer
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberDrawerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.example.reply.R
+import com.example.reply.ui.DevicePosture
+import com.example.reply.ui.ReplyContentType
+import com.example.reply.ui.ReplyDestinations
+import com.example.reply.ui.ReplyHomeUIState
+import com.example.reply.ui.ReplyNavigationType
+import com.example.reply.ui.WindowSize
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReplyApp(
+    windowSize: WindowSize,
+    foldingDevicePosture: DevicePosture,
+    replyHomeUIState: ReplyHomeUIState
+) {
+    /**
+     * This will help us select type of navigation and content type depending on window size and
+     * fold state of the device.
+     *
+     * In the state of folding device If it's half fold in BookPosture we want to avoid content
+     * at the crease/hinge
+     */
+    val navigationType: ReplyNavigationType
+    val contentType: ReplyContentType
+    when (windowSize) {
+        WindowSize.COMPACT -> {
+            navigationType = ReplyNavigationType.BOTTOM_NAVIGATION
+            contentType = ReplyContentType.LIST_ONLY
+        }
+        WindowSize.MEDIUM -> {
+            navigationType = ReplyNavigationType.NAVIGATION_RAIL
+            contentType = if (foldingDevicePosture != DevicePosture.NormalPosture) {
+                ReplyContentType.LIST_AND_DETAIL
+            } else {
+                ReplyContentType.LIST_ONLY
+            }
+        }
+        WindowSize.EXPANDED -> {
+            navigationType = if (foldingDevicePosture is DevicePosture.BookPosture) {
+                ReplyNavigationType.NAVIGATION_RAIL
+            } else {
+                ReplyNavigationType.PERMANENT_NAVIGATION_DRAWER
+            }
+            contentType = ReplyContentType.LIST_AND_DETAIL
+        }
+    }
+
+    val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
+    val scope = rememberCoroutineScope()
+
+    val selectedDestination = ReplyDestinations.INBOX
+
+    if (navigationType == ReplyNavigationType.PERMANENT_NAVIGATION_DRAWER) {
```

Comment:
can we extract the navigation to a different function? right now the composable is quite long and hard to read